### PR TITLE
Added ReferenceArraySetLength method.

### PR DESCRIPTION
**Pull request by phx-mpascale**

### CODE REVIEW

**Review by cleftwich — APPROVED** [review]

File: ansys/api/modelcenter/v0/workflow_messages.proto [hunk 1]
```diff
@@ -453,3 +453,25 @@ message MoveComponentRequest {
...
+message SetReferenceArrayLengthRequest {
+  /**
+   * The reference array to modify.
+   */
+  ansys.api.modelcenter.v0.element_messages.ElementId target = 1;
+
+  /**
+   * The new size of the array.
+   */
+  ansys.api.modelcenter.v0.variable_value_messages.IntegerValue new_size = 2;
```
Comment (phx-mkoninckx) [inline ansys/api/modelcenter/v0/workflow_messages.proto:469]: Why not just use one of gRPC's built in integer primitives here?

File: ansys/api/modelcenter/v0/grpc_modelcenter_workflow.proto [hunk 2]
```diff
@@ -833,6 +833,14 @@ service ModelCenterWorkflowService {
   rpc ReferenceArrayGetLength(ansys.api.modelcenter.v0.element_messages.ElementId)
     returns (ansys.api.modelcenter.v0.variable_value_messages.IntegerValue);
 
+  /**
+   * Set the length of a reference array.
+   * If larger that the current size, the array will be expanded with empty elements.
+   * If smaller than the current size, items at the end of the array will be dropped.
```
Comment (phx-mkoninckx) [inline ansys/api/modelcenter/v0/grpc_modelcenter_workflow.proto:839]: May want to document what error you'll get back if you pass an invalid (e.g. negative) size (`OUT_OF_RANGE` seems reasonable to me).

**Review by phx-mkoninckx — CHANGES_REQUESTED** [review]

File: ansys/api/modelcenter/v0/workflow_messages.proto (hunk 1 above)
Comment (phx-mpascale) [reply ansys/api/modelcenter/v0/workflow_messages.proto:469]: Hm, I was just mirroring the get size method, but I could switch it if we don't think it makes sense here.

Comment (phx-mkoninckx) [reply ansys/api/modelcenter/v0/workflow_messages.proto:469]: But here you are already in a message type, so you're free to use the primitives if you want.

**Review by ansggarcia — APPROVED** [review]

**Review by phx-mkoninckx — APPROVED** [review]

**Pull request merged**